patches/net-proxy_resolution-proxy_resolution_service.cc.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/net/proxy_resolution/proxy_resolution_service.cc b/net/proxy_resolution/proxy_resolution_service.cc
+index 0cb73b2594ab13cfcbf26a37aca4f8a7da6b7b02..16a8c509d9aca9a458b384c4fc98c322ed0113d0 100644
+--- a/net/proxy_resolution/proxy_resolution_service.cc
++++ b/net/proxy_resolution/proxy_resolution_service.cc
+@@ -1181,6 +1181,9 @@ int ProxyResolutionService::ResolveProxy(const GURL& raw_url,
+   if (current_state_ == STATE_NONE)
+     ApplyProxyConfigIfAvailable();
+ 
++  if (IsTorProxy(config_->value()) && raw_url.ref() == "NewTorCircuit")
```

Comment:
IMO we should hide this under `ENABLE_TOR`, though not sure if it's possible

Reply:
We try to make //net not to rely on any brave stuff to avoid circular dependency.

Reply:
please update patches similar to discussions about sync patches

Reply:
actually I'd really like to find some way to not have tor-specific stuff in ProxyResolutionService, in particular the tor_proxy_map_. Can we create a class a singleton class in Brave that keeps a map of `ProxyResolutionService` instance -> `tor_proxy_map` and use that inside a chromium_src injected method/macro?